installation/routines/setup_autohotspot.sh

```diff
@@ -1,119 +1,91 @@
-    sudo cp "${INSTALLATION_PATH}"/resources/autohotspot/autohotspot.timer "${cron_autohotspot_file}"
-    sudo sed -i "s|%%USER%%|${CURRENT_USER}|g" "${cron_autohotspot_file}"
+_config_file_backup() {
+    # create flag file or copy present conf to orig file
+    # to correctly handling future deactivation of autohotspot
+    local config_file="$1"
+    local config_flag_file="${config_file}.remove"
+    local config_orig_file="${config_file}.orig"
+    if [ ! -f "${config_file}" ]; then
+        sudo touch "${config_flag_file}"
+    elif [ ! -f "${config_orig_file}" ] && [ ! -f "${config_flag_file}" ]; then
+        sudo cp "${config_file}" "${config_orig_file}"
+    fi
 }
 
-_install_autohotspot_script() {
-    sudo cp "${INSTALLATION_PATH}"/resources/autohotspot/autohotspot "${AUTOHOTSPOT_TARGET_PATH}"
-    sudo chmod +x "${AUTOHOTSPOT_TARGET_PATH}"
+_config_file_revert() {
+    # revert config files to original (remove if it wasn't existing before)
+    local config_file="$1"
+    local config_flag_file="${config_file}.remove"
+    local config_orig_file="${config_file}.orig"
+    if [ -f "${config_flag_file}" ]; then
+        sudo rm "${config_flag_file}" "${config_file}"
+    elif [ -f "${config_orig_file}" ]; then
+        sudo mv "${config_orig_file}" "${config_file}"
+    fi
 }
 
-
-_autohotspot_check() {
-    print_verify_installation
-
-    verify_apt_packages hostapd dnsmasq iw
-
-    verify_service_enablement hostapd.service disabled
-    verify_service_enablement dnsmasq.service disabled
-    verify_service_enablement autohotspot.service enabled
-
-    verify_files_exists "/etc/cron.d/autohotspot"
-    verify_files_exists "${AUTOHOTSPOT_TARGET_PATH}"
-
-    verify_file_contains_string "${WIFI_INTERFACE}" "${AUTOHOTSPOT_HOSTAPD_CONF_FILE}"
-    verify_file_contains_string "${AUTOHOTSPOT_PASSWORD}" "${AUTOHOTSPOT_HOSTAPD_CONF_FILE}"
-    verify_file_contains_string "${WIFI_REGION}" "${AUTOHOTSPOT_HOSTAPD_CONF_FILE}"
-    verify_file_contains_string "${AUTOHOTSPOT_HOSTAPD_CONF_FILE}" "${AUTOHOTSPOT_HOSTAPD_DAEMON_CONF_FILE}"
-
-    verify_file_contains_string "${WIFI_INTERFACE}" "${AUTOHOTSPOT_DNSMASQ_CONF_FILE}"
-    verify_file_contains_string "nohook wpa_supplicant" "${AUTOHOTSPOT_DHCPD_CONF_FILE}"
+_get_last_ip_segment() {
+    local ip="$1"
+    echo $ip | cut -d'.' -f1-3
 }
 
-_run_setup_autohotspot() {
-    _install_packages
-    _get_interface
-    _configure_hostapd
-    _configure_dnsmasq
-    _other_configuration
-    _install_autohotspot_script
-    _install_service_and_timer
-    _autohotspot_check
-}
 
 setup_autohotspot() {
     if [ "$ENABLE_AUTOHOTSPOT" == true ] ; then
-        run_with_log_frame _run_setup_autohotspot "Install AutoHotspot"
+        local installed=false
+        if [[ $(_is_dhcpcd_enabled) == true || "${CI_RUNNING}" == "true" ]]; then
+            run_with_log_frame _run_setup_autohotspot_dhcpcd "Install AutoHotspot dhcpcd"
+            installed=true
+        fi
```

Comment:
Do we have this only for backwards compatibility? I don't really understand why we want to support dhcpcd and NetworkManager.

To reduce complexity, why don't we migrate to NetworkManager directly and ignore dhcpcd directly. Or do we need this for Bullseye, because it does not support NetworkManager?

Reply:
Its for compatibility.
Bullseye supports NetworkManager, but its not default.
The user can change it (similar to the description in https://github.com/MiczFlor/RPi-Jukebox-RFID/issues/2184), but its a manual task.

It also not trivial to automate it, as the network will reset and the connection will be lost.
Also there would be a mapper from WPA entries needed to activate wifi again, if the user has only wifi connection (which will be in most cases i suppose).